src/commands/git/worktree.ts

```diff
@@ -317,7 +319,24 @@ export class WorktreeGitCommand extends QuickCommand<State> {
 						state.uris = [state.uris];
 					}
 
-					yield* this.deleteCommandSteps(state as DeleteStepState, context);
+					const branchesOfDeletedWorktrees = yield* this.deleteCommandSteps(
+						state as DeleteStepState,
+						context,
+					);
+					if (branchesOfDeletedWorktrees !== StepResultBreak && branchesOfDeletedWorktrees?.length) {
+						yield* getSteps(
+							this.container,
+							{
+								command: 'branch',
+								state: {
+									subcommand: 'delete',
+									repo: state.repo,
+									references: branchesOfDeletedWorktrees,
+								},
+							},
+							this.pickedVia,
+						);
+					}
```

Comment:
Could we move this into `this.deleteCommandSteps`? I tried it locally and it seems fine.